### PR TITLE
remove double-splat from options param, puma.rb

### PR DESCRIPTION
**Pull request by AJ8GH**
### Description
Please describe your pull request. Thank you for contributing! You're the best.
- remove double-splat from options parameter, in puma.rb line 63
- was throwing argument error when running sinatra and puma with shotgun
- all existing tests still pass after removing and the bug is fixed - shotgun now raises no error and works as expected

### Your checklist for this pull request

- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added an entry to [History.md](../blob/master/History.md) if this PR fixes a bug or adds a feature. If it doesn't need an entry to HISTORY.md, I have added `[changelog skip]` or `[ci skip]` to the pull request title.
- [ ] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [ ] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: History.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 5.2.2 / 2021-02-15
+
+* Bugfixes
+  * Remove `**` from `options` parameter ([#2555])
+
 ## 5.2.2 / 2021-02-
 
 * Bugfixes
```
Comment (AJ8GH) [inline History.md:8]: this was causing puma to throw argument error when running sinatra with shotgun
removing double splat from options param fixes the issue, shogun runs fine with puma and sinatra
all tests still passing after making the fix

**Comment by dentarg** [comment]
> * was throwing argument error when running sinatra and puma with shotgun

Can you provide steps to reproduce?

**Comment by AJ8GH** [comment]
Sure, this should reproduce the error:

create project with Gemfile:
`mkdir puma-bug-test`
`cd puma-bug-test`
`bundle init`

add gems to Gemfile:
```
gem 'puma'
gem 'shotgun'
gem 'sinatra'
```

bundle install:
`bundle install`

create bug test file:
`touch bug_test.rb`

add sinatra/ruby code to bug test file:
```
require 'sinatra'
get '/' do
  'hello world'
end
```

run file using shotgun:
`shotgun bug_test.rb -p 4567`

**Comment by dentarg** [comment]
@AJ8GH Thanks for reporting, but this is not a problem that is fixed in Puma, but instead in https://github.com/rtomayko/shotgun (which sadly hasn't been updated since 2016, so chances for that seem slim; I can recommend https://github.com/alexch/rerun, it's at least alive :-))

---

Using Ruby 3.0, we can see it fails like this:

```
$ bundle e shotgun test.rb                             
== Shotgun/Puma on http://127.0.0.1:9393/
bundler: failed to load command: shotgun (/Users/dentarg/.gem/ruby/3.0.0/bin/shotgun)
/Users/dentarg/.gem/ruby/3.0.0/gems/puma-5.2.1/lib/rack/handler/puma.rb:63:in `run': wrong number of arguments (given 2, expected 1) (ArgumentError)
	from /Users/dentarg/.gem/ruby/3.0.0/gems/shotgun-0.9.2/bin/shotgun:156:in `<top (required)>'
	from /Users/dentarg/.gem/ruby/3.0.0/bin/shotgun:23:in `load'
	from /Users/dentarg/.gem/ruby/3.0.0/bin/shotgun:23:in `<top (required)>'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/site_ruby/3.0.0/bundler/cli/exec.rb:63:in `load'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/site_ruby/3.0.0/bundler/cli/exec.rb:63:in `kernel_load'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/site_ruby/3.0.0/bundler/cli/exec.rb:28:in `run'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/site_ruby/3.0.0/bundler/cli.rb:494:in `exec'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/site_ruby/3.0.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/site_ruby/3.0.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/site_ruby/3.0.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/site_ruby/3.0.0/bundler/cli.rb:30:in `dispatch'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/site_ruby/3.0.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/site_ruby/3.0.0/bundler/cli.rb:24:in `start'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/bundler-2.2.4/exe/bundle:49:in `block in <top (required)>'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/site_ruby/3.0.0/bundler/friendly_errors.rb:130:in `with_friendly_errors'
	from /Users/dentarg/.rubies/ruby-3.0.0/lib/ruby/gems/3.0.0/gems/bundler-2.2.4/exe/bundle:37:in `<top (required)>'
	from /Users/dentarg/.gem/ruby/3.0.0/bin/bundle:23:in `load'
	from /Users/dentarg/.gem/ruby/3.0.0/bin/bundle:23:in `<main>'
```

Using Ruby 2.7.1, it still works and we can see the warnings:

```
$ bundle e shotgun test.rb 
== Shotgun/Puma on http://127.0.0.1:9393/
/Users/dentarg/.gem/ruby/2.7.1/gems/shotgun-0.9.2/bin/shotgun:156: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/Users/dentarg/.gem/ruby/2.7.1/gems/puma-5.2.1/lib/rack/handler/puma.rb:63: warning: The called method `run' is defined here
Puma starting in single mode...
* Puma version: 5.2.1 (ruby 2.7.1-p83) ("Fettisdagsbulle")
*  Min threads: 0
*  Max threads: 5
*  Environment: development
*          PID: 55809
* Listening on http://127.0.0.1:9393
Use Ctrl-C to stop
^C- Gracefully stopping, waiting for requests to finish
=== puma shutdown: 2021-02-16 00:06:27 +0100 ===
- Goodbye!
```

Using the change in https://github.com/rtomayko/shotgun/pull/74, it is possible to use Ruby 3.0 and shotgun:

```
$ ruby -e 'p RUBY_VERSION' && bundle e shotgun test.rb
"3.0.0"
== Shotgun/Puma on http://127.0.0.1:9393/
Puma starting in single mode...
* Puma version: 5.2.1 (ruby 3.0.0-p0) ("Fettisdagsbulle")
*  Min threads: 0
*  Max threads: 5
*  Environment: development
*          PID: 64519
* Listening on http://127.0.0.1:9393
Use Ctrl-C to stop
```

**Comment by dentarg** [comment]
Related PR: #2189

**Comment by AJ8GH** [comment]
Ok, cool thank you! I reverted the changes I made to my puma file and added the ** to shotgun instead.

**Pull request closed without merging**